G-Research.OpenTelemetry.Processor.Partial/PartialActivityProcessor.cs

```diff
@@ -208,37 +147,40 @@ protected override bool OnShutdown(int timeoutMilliseconds)
         {
             case Timeout.Infinite:
                 exporterThread.Join();
-                return logExporter.Shutdown();
+                return logExporter.Shutdown() && logProcessor.Shutdown();
             case 0:
-                return logExporter.Shutdown(0);
+                return logExporter.Shutdown(0) && logProcessor.Shutdown(0);
         }
 
         var sw = Stopwatch.StartNew();
         exporterThread.Join(timeoutMilliseconds);
         var timeout = timeoutMilliseconds - sw.ElapsedMilliseconds;
-        return logExporter.Shutdown((int)Math.Max(timeout, 0));
+        return logExporter.Shutdown((int)Math.Max(timeout, 0)) &&
+               logProcessor.Shutdown((int)Math.Max(timeout, 0));
     }
 
     protected override void Dispose(bool disposing)
     {
         base.Dispose(disposing);
         shutdownTrigger.Dispose();
+        logProcessor.Dispose();
+        loggerFactory.Dispose();
     }
 
-    private static List<KeyValuePair<string, object?>> GetLogRecordAttributes() =>
-    [
-        new("telemetry.logs.cluster", "partial"),
-        new("telemetry.logs.project", "span"),
-    ];
-
-    private List<KeyValuePair<string, object?>> GetHeartbeatLogRecordAttributes() =>
-    [
-        new("partial.event", "heartbeat"),
-        new("partial.frequency", heartbeatIntervalMilliseconds + "ms")
-    ];
-
-    private static List<KeyValuePair<string, object?>> GetStopLogRecordAttributes() =>
-    [
-        new("partial.event", "stop"),
-    ];
+    private Dictionary<string, object> GetHeartbeatLogRecordAttributes() => new()
+    {
+        ["partial.event"] = "heartbeat",
+        ["partial.frequency"] = $"{heartbeatIntervalMilliseconds}ms",
+        ["partial.body.type"] = "json",
```

Comment:
Include a version number so we can potentially change the json schema in future.

```suggestion
        ["partial.body.type"] = "json/v1",
```